File: PropertyRenting.Api/Contracts/Requests/Contract.cs
using PropertyRenting.Api.Enums;

namespace PropertyRenting.Api.Contracts.Requests
{
    public record AddRequest(int ContractNumber, string BuildingId, string OwnerId, string Description, DateTime ContractDate, DateTime ContractStartDate, DateTime ContractEndDate, decimal ContractAmount, PaymentMethod PaymentMethod);
    public record UpdateRequest(string Id, int ContractNumber, string BuildingId, string OwnerId, string Description, DateTime ContractDate, DateTime ContractStartDate, DateTime ContractEndDate, decimal ContractAmount, PaymentMethod PaymentMethod);
    public record DeleteRequest(string Id);

    public record GetRequest(string Id);
    public record GetAllRequest();
    public record SearchRequest();
    public record GetPageRequest(int PageNumber, int PageSize);
}

File: PropertyRenting.Api/ClientApp/src/app/Models/contract.ts
export interface Contract {
    id?: any;
    contractNumber?: number;
    buildingId?: any;
    ownerId?: any;
    description?: string;
    contractDate?: Date;
    contractStartDate?: Date;
    contractEndDate?: Date;
    contractAmount?: number;
    contractState?: number;
    paymentMethod?: number;
}
